api/src/opentrons/protocol_runner/legacy_command_mapper.py

```diff
@@ -192,78 +229,223 @@ def _build_initial_command(
-                raise Exception("Unknown pick_up_tip location.")
+            engine_command = self._build_liquid_handling_commands(
+                command=command, command_id=command_id, now=now
+            )
+        elif command["name"] == legacy_command_types.BLOW_OUT:
+            engine_command = self._build_blow_out_command(
+                command=command, command_id=command_id, now=now
+            )
+        elif command["name"] == legacy_command_types.PAUSE:
+            engine_command = pe_commands.WaitForResume.construct(
+                id=command_id,
+                key=command_id,
+                status=pe_commands.CommandStatus.RUNNING,
+                createdAt=now,
+                startedAt=now,
+                params=pe_commands.WaitForResumeParams.construct(
+                    message=command["payload"]["userMessage"],
+                ),
+            )
+        else:
+            engine_command = pe_commands.Custom.construct(
+                id=command_id,
+                key=command_id,
+                status=pe_commands.CommandStatus.RUNNING,
+                createdAt=now,
+                startedAt=now,
+                params=LegacyCommandParams.construct(
+                    legacyCommandType=command["name"],
+                    legacyCommandText=command["payload"]["text"],
+                ),
+            )
+
+        return engine_command
+
+    def _build_drop_tip_command(
+        self,
+        command: legacy_command_types.DropTipMessage,
+        command_id: str,
+        now: datetime,
+    ) -> pe_commands.Command:
+        pipette: LegacyPipetteContext = command["payload"]["instrument"]
+        location = command["payload"]["location"]
+        if location.labware.is_well:
+            well = location.labware.as_well()
+        mount = MountType(pipette.mount)
+        #   the following type checking suppression assumes the tiprack is not loaded on top of a module
+        slot = DeckSlotName.from_primitive(well.parent.parent)  # type: ignore[arg-type]
+        well_name = well.well_name
+        labware_id = self._labware_id_by_slot[slot]
+        pipette_id = self._pipette_id_by_mount[mount]
+        return pe_commands.DropTip.construct(
+            id=command_id,
+            key=command_id,
+            status=pe_commands.CommandStatus.RUNNING,
+            createdAt=now,
+            startedAt=now,
+            params=pe_commands.DropTipParams.construct(
+                pipetteId=pipette_id,
+                labwareId=labware_id,
+                wellName=well_name,
+            ),
+        )
+
+    def _build_pick_up_tip_command(
+        self,
+        command: legacy_command_types.PickUpTipMessage,
+        command_id: str,
+        now: datetime,
+    ) -> pe_commands.Command:
+        pipette: LegacyPipetteContext = command["payload"]["instrument"]
+        location = command["payload"]["location"]
+        if isinstance(location, LegacyWell):
+            well = location
```

Comment:
Same comment here; the PAPI is supposed to guarantee this, so we should assert rather than branch

```suggestion
        assert isinstance(location, LegacyWell)
        well = location
```